packages/jest-expect/src/types.ts

```diff
@@ -14,8 +14,8 @@ import type {
 
 export type JestExpect = {
   <T = unknown>(actual: T): JestMatchers<void, T> &
-    Inverse<JestMatchers<void, T>> &
-    PromiseMatchers<T>;
+    Inverse<JestMatchers<void, unknown>> &
+    PromiseMatchers<unknown>;
```

Comment:
Hm.. This looks unexpected.

Reply:
it avoids type checking in these matchers, as it would break existing tests